test/e2e_node/resource_metrics_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/kubelet/apis/resourcemetrics/v1alpha1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/framework/metrics"
+
+	"github.com/prometheus/common/model"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gstruct"
+	"github.com/onsi/gomega/types"
+)
+
+const (
+	pod0        = "stats-busybox-0"
+	pod1        = "stats-busybox-1"
+	maxStatsAge = time.Minute
+)
+
+var _ = framework.KubeDescribe("ResourceMetricsAPI", func() {
+	f := framework.NewDefaultFramework("resource-metrics")
+	Context("when querying /resource/metrics", func() {
+		BeforeEach(func() {
+			By("Creating test pods")
+			numRestarts := int32(1)
+			pods := getSummaryTestPods(f, numRestarts, pod0, pod1)
+			f.PodClient().CreateBatch(pods)
+
+			Eventually(func() error {
+				for _, pod := range pods {
+					err := verifyPodRestartCount(f, pod.Name, len(pod.Spec.Containers), numRestarts)
+					if err != nil {
+						return err
+					}
+				}
+				return nil
+			}, time.Minute, 5*time.Second).Should(BeNil())
+
+			// Wait for cAdvisor to collect 2 stats points
+			time.Sleep(15 * time.Second)
+		})
+		It("should report resource usage through the v1alpha1 resouce metrics api", func() {
+			// fetch node so we can know proper node memory bounds for unconstrained cgroups
```

Comment:
Favor `By` lines to comments here (makes it easier to trace failures down to the part that failed)